### ISSUE TITLE
Unable to SIgn In via Github

### ISSUE THREAD

**Issue by captainkovalsky** [stackblitz/core#1402]
I see this one

<img width="828" alt="Screenshot 2021-03-12 at 19 15 32" src="https://user-images.githubusercontent.com/845156/110974655-6be56d00-8367-11eb-88b2-159d974d7a6e.png">

My GitHub info is https://api.github.com/users/captainkovalsky

link https://stackblitz.com/api/users/auth/github/callback?code=5cb672be49040b6158e3&state=e59ccd7889fe1968fb49b24b894c76adc7609f9f318ea5b3

![image](https://user-images.githubusercontent.com/845156/110988733-9cce9d80-8379-11eb-93fc-f4cb3658a6ce.png)

In Private Mode the same 
Browsers: Microsoft Edge, Chrome

[stackblitz.com.har.zip](https://github.com/stackblitz/core/files/6132579/stackblitz.com.har.zip) - for better checking what's going on with redirect auth

Possible root cause:
I had two github accounts. First captainkovalsky and second furydam (nicknames)

after transferring all repositories from captainkovalsky to furydam I've removed captainkovalsky account and rename furydam to captainkovalsky. So possibly that's the reason why I can not login.

Possibly related to the #1383

**Comment by kumardevsingh** [comment]
I have same problem. Please help me.

**Comment by kumardevsingh** [comment]
any update

**Comment by SamVerschueren** [comment]
So when this happens, it's likely that it's because of an email or username collision. The error message isn't helpful and we need to work on that.

**So what does this mean?**
It happens if for instance someone creates two GitHub accounts with the same email address. So account A and account B have the same email address. Someone signs in with account A at StackBlitz. Later that person signs in with account B, then we have a collision.

It can also happen with a username. If account A has username Foo, signs in at StackBlitz. Then account A gets renamed to FooBar and a second account B is being created with username Foo. If account B signs in, we know have a username collision because that username is taken by someone else.

**How to resolve?**
If it's an email collision, you have two options 
1. Update email at Account B to have a +github added or something so the email is different from Account A.
2. Update email at Account A to something else, sign in with Account A at StackBlitz so the information in our database gets updated. Afterwards, sign in with Account B and now it should work.

If it's a username collision, it probably depends. If some random person once was in possession of your current GitHub username and signed in at StackBlitz, we will have to fix it manually on our side. If you are in possession of both GitHub accounts, it's basically the same as solution 2 of the email collision flow. Sign in with Account A so the info gets updated on our side, then sign in with account B and it should work.

Hopefully this resolves the issue for both of you. If not, don't hesitate to ask and we'll have a closer look.

**Comment by m8xp0w3r** [comment]
@SamVerschueren I have a different scenario. I owned a private github account and an account to use within the company I work for. Now I transferred my private to my business account and deleted the obsolete ex-private account. So I could not follow your guide, because the second account is deleted. My only account now owns both email adresses (private and business). How can I solve this issue?

**Comment by SamVerschueren** [comment]
I believe we have to remove your private account (which you removed) on our end. Is it a username or email collision in your case? The more information we have, the better we can resolve.

**Comment by m8xp0w3r** [comment]
@SamVerschueren I don't get from the error message if it is an email or user conflict. Removing my old account from stackblitz would be okay for me. I do not have important projects there. Until now I only used this as as playground. Where can I provide you the needed information to delete this account?

**Comment by SamVerschueren** [comment]
Yes we definitely need to improve the error messages when running in this issue. If you don't have a way of signing in into your old account and resolve it yourself, you can put your old username here and I'll get it fixed.

**Comment by captainkovalsky** [comment]
I understand why I got the login error.

I had two github accounts. First captainkovalsky  and second furydam (nicknames)

after transferring  all repositories from captainkovalsky to furydam I've removed captainkovalsky account and rename furydam to captainkovalsky. So possibly that's the reason why I can not login. What should I do in this case ?

**Comment by ponich** [comment]
Did someone resolve it?

**Comment by SamVerschueren** [comment]
I have a PR ready to improve the error message when the duplicate username happens. Will let you all know when that lands so you can check if that's the issue in your case.

Everyone who is not able to resolve this on their own, respond in the thread here indicating that we have to resolve this on our end and I'll make it happen.

**Comment by derekowusubekoe** [comment]
>     1. github

I have checked my account and I have only one email.

**Comment by captainkovalsky** [comment]
I've three emails, tried each of them and still unable to login.

**Comment by m8xp0w3r** [comment]
> Yes we definitely need to improve the error messages when running in this issue. If you don't have a way of signing in into your old account and resolve it yourself, you can put your old username here and I'll get it fixed.

@SamVerschueren maybe my old account was with this username: djlatscho my email from this account is/was: nilsleistner@me.com

Would be great, if this gets deleted that I can start using stackblitz again.

**Comment by SamVerschueren** [comment]
The error message should be better now when signing in while hitting this problem. I believe it was released last Thursday. 

From my understanding it's almost always a duplicate username issue.

As I said before, if you're unable to resolve it yourself, make a comment clearly indicating that we should solve it on our end and and what GitHub account is affected.

**Comment by captainkovalsky** [comment]
this username: captainkovalsky
account email is/was: dzundzaviktr@gmail.com

**Comment by m8xp0w3r** [comment]
@SamVerschueren I am still getting this error message:

<img width="842" alt="image" src="https://user-images.githubusercontent.com/38067181/112950020-1d431b80-913a-11eb-966b-1e4cee96e52c.png">

actual username: m8xp0w3r
account email: nilsleistner@me.com / nils.leistner@cn-consult.eu

**Comment by captainkovalsky** [comment]
btw, is it possible to fix this issue in some different way?

**Comment by m8xp0w3r** [comment]
@SamVerschueren because I still can't log in, I want to renew my request: I can't solve this problem and I am still getting the error message above in my last comment. Your enhanced version of this error message, you are speaking above, is not coming to me.
Here is my current github profile data:
actual username: m8xp0w3r
account email: nilsleistner@me.com / nils.leistner@cn-consult.eu

Background for my request: In our company we are now using Angular and I wanted to test if stackblitz could be a nice addon for normal desktop developing. As long as I can't log in, this test is not possible, because of private repos. 😭

**Comment by m8xp0w3r** [comment]
@captainkovalsky is your problem solved? Can you log in again?

**Comment by captainkovalsky** [comment]
@m8xp0w3r tried to login, see the error.
My profile name: captainkovalsky
emails: dzundzaviktr@gmail.com / victor.dzundza.dev@gmail.com

**Comment by captainkovalsky** [comment]
I've read the topic #1421. My new github username is exactly the same as old username (I've renamed it) so need to remove my stackblitz account, please

**Comment by SamVerschueren** [comment]
Sorry that I didn't get back earlier. I still have to process the requests here but I'm waiting for a feature to land which makes it easier to do this. When that happens I'll keep you guys in the loop 🙏 .

**Comment by SamVerschueren** [comment]
@captainkovalsky @m8xp0w3r I forwarded your request but we have a procedure to go through because it involves deleting data.

For all other people here having the exact same issue. Please respond with your github username and email address or addresses. The more info we have, te better we can help. If you know why this error happens (because of some username renaming for instance), even better!

**Comment by captainkovalsky** [comment]
Thank you @SamVerschueren 

I've tried to login and everything is fine for me .

**Comment by m8xp0w3r** [comment]
Same for me. Thank you @SamVerschueren for your help.

**Comment by SamVerschueren** [comment]
Great to see that this works now for you guys 🙏 !

I'm going to close the issue. For anyone else, feel free to open a new issue with the required information and I'll make sure to forward the request.

**Comment by kc0tlh** [comment]
**Update on this issue as of 3/23/2022**

There are at least 3 known causes of the Github Auth Login Issue:

1. You may not have a confirmed email address in Github. You can ensure your email address is verified on Github's email settings page to resolve this issue:​ https://github.com/settings/emails

2. You have changed your Github username or email address since you last logged into StackBlitz, causing a collision in our database. 

3. You have two separate GitHub accounts with an overlapping email address and/or username. 

If you are in situation 2 or 3, a temporary workaround is to change the email address and/or username of the Github account that you would like to log into StackBlitz with to resolve the collision in our database. 

Tip: If you don't want to change your actual Github email address, you can instead update it to have a +github added to the end, so it is unique. For example [alex@gmail.com](mailto:alex@gmail.com) can be entered into github as [alex+github@gmail.com](mailto:alex+github@gmail.com) and all email will still flow to the existing "alex" Gmail account.

Hopefully this temporary workaround resolves the issue for you. If not, please open a new issue, reference #1402, and provide as much detail as possible (what was your Github username and email address before, what is it now, etc) and we will take a look!

We are working internally to resolve this issue in a more permanent and clean way. Follow this issue for updates!

**Comment by purplem1lk** [comment]
**Update on this issue as of July 11, 2022:**

Thanks all for your patience on this! We tested and pushed fixes for the 3 known causes of the GitHub Auth Login Issue, which should be permanent fixes so you shouldn’t need to use any previous workarounds listed in our previous update.

If you run into any login issues, please open a new issue, reference [#1402](https://github.com/stackblitz/core/issues/1402), and provide as much detail as possible (what was your Github username and email address before, what is it now, etc) and we will take a look!
